~~D'oh, I was in an older env (npm 6.2.0) where it does fail. Using a newer version, I too see that scopes are supported.~~

I didn't look at this carefully enough as it does work in npm 6.2.0. I thought I had tried it a few different ways and that it wasn't working. For example doing something like:
❯ npm repo @angular
npm ERR! code EINVALIDTAGNAME
npm ERR! Invalid tag name "@angular": Tags may not have any characters that encodeURIComponent encodes.

I originally read this as meaning that @ is an invalid character. Anyway, scratch what I said about it not working.

--- a/plugin/github_link_opener.vim
+++ b/plugin/github_link_opener.vim
@@ -17,7 +17,8 @@ function! s:OpenGitHubLink()
     let url = matchstr(word, chars . '/' . chars . '/' . chars)
     call s:OpenWithNetrw("https://" . url)
   elseif index(['javascript', 'javascript.jsx', 'typescript'], &ft) >= 0 && line_has_js_package
-    let package = matchstr(getline('.'), '\v[''"]\@?\zs(\w+[-.]?)+\ze')
+    let re_scoped_or_nonscoped_npm_package = '\v[''"]\zs((\@(\w+[-.]?)+\/(\w+[-.]?)+)|(\w+[-.]?)+)'
+    let package = matchstr(getline('.'), re_scoped_or_nonscoped_npm_package)
     if !empty(package)
       silent execute '!npm repo ' . package
     endif

